### DISCUSSION TITLE
Is there a way to have f2b ignore additional log errors within N seconds of one that causes a ban?

### DISCUSSION THREAD

**Discussion by MrPeteH** [fail2ban/fail2ban#3569] [Q&A] [answered]
I'm seeing a common pattern where a server process has created multiple lines of evidence against a bad actor. The log file thus contains multiple entries within a few seconds, any/all that will cause a ban. The result:
1) f2b bans on the first bad log line
2) f2b sees another, and says "<IP> already banned"

Seems there ought to be a way for that to not happen within the same log timeframe of N seconds?

**Comment by sebres** [comment]
> Seems there ought to be a way for that to not happen within the same log timeframe of N seconds?

It was planned, and partially implemented somewhere in v.1.0 (or in some experimental branch that is not yet merged), but IIRC it does that in the way, that messages get ignored some short time after ban, but only failures which logged before ban (whose time is smaller than time of ban). 
This "feature" is relative new and may not work correct (if it is merged/released at all).

Note that "already banned" can still mistakenly happen also by another conditions:
- some messages are retarded (e. g. waiting for response to log its code) and therefore logged delayed;
- a certain delay in net-filter subsystem after ban (or some whitelisting rules in firewall allowing established connections) unless the connections get rejected;

**Comment by MrPeteH** [comment, marked as answer]
> Note that "already banned" can still mistakenly happen also by another conditions...
I agree 100%. To me this is a generalized issue.

Perhaps the best resolution is a no-code solution: highlight in documentation that "Already Banned" is not necessarily a problem at all.

**Reply by sebres** [reply]
But it is or rather it depends - if it'd continuously log "already banned", also several minutes after ban, it may signal that something is wrong either with banning action (doesn't ban at all) or with config (missed some port/proto/etc) of with net-filter (some whitelisting rules before f2b chains, e.g. incorrect precedence etc).
